### ISSUE TITLE
Client and server mismatch

### ISSUE THREAD

**Issue by tomgreeEn** [vinissimus/next-translate#209]
Hi
I'm using the build solution and everything appears to work OK except that I am getting an error that the client and server don't match:

react_devtools_backend.js:6 Warning: Prop `href` did not match. Server: "/products" Client: "/en/products". This is my setup (currently only using one language):

"allLanguages": ["en"],
  "defaultLanguage": "en",
  "currentPagesDir": "pages_",
  "finalPagesDir": "pages",
  "localesPath": "locales",
  "defaultLangRedirect": "root",

As I'm using defaultLangRedirect, I don't see the /en on the href attribute on the client - could it be that on first render it has this attribute before it is corrected to the default language subsequently?

**Comment by aralroca** [comment]
Are you using `Link` component from `next-translate/Link`? Could you share your `/products` page code?

Currently in our [example](https://github.com/vinissimus/next-translate/tree/master/examples/static-site), we also are using `"defaultLangRedirect": "root"` together with `<Link />` and we don't have this warning.

Which version of `next-translate` & `next` are you using?

**Comment by tomgreeEn** [comment]
Thanks for the quick reply. I actually realised I was using i18nprovider in a component wrapper higher in the tree in _app, which may not be recommended usage so I've switched it off for now while I try to get my head around options here.

However, I still have one outstanding issue inside Component with next-translate/Router:

Router.pushI18n( '/products/[...products]', '/products/running-shoes/2', {shallow: true});
  // tries to load '//products/running-shoes/2' -> should load '/products/running-shoes/2'

If I switch to   "defaultLangRedirect": "lang-path" I get the same result, and also in all combinations of adding /en to the front of the href and/or as path, as well as removing the preceding slashes, though I believe they should be there. The documentation doesn't include any examples using 'as', is this supported?

Versions: 
    "next": "^9.3.5",
   "next-translate": "^0.16.1",
